tools/topology/topology2/platform/intel/dmic-generic.conf

```diff
@@ -40,7 +40,7 @@ Object.Dai.DMIC [
                dai_index               1
                driver_version          $DMIC_DRIVER_VERSION
                io_clk                  $DMIC_IO_CLK
-               sample_rate             16000
+	       sample_rate             $DMIC1_RATE
```

Comment:
commit message: 

"
The sample rate of DMIC1 DAI can be set to 8 - 96 kHz with
macro DMIC1_RATE.
"

I thought DMIC1_RATE had to be lower than DMIC0_RATE? Are the two independent?

Reply:
Yes, the DMIC0 and DMIC1 FIR are identical, they share the same CIC. There can be some restrictions for rates from DMIC bus clock constraints (min, max Hz, duty cycle) and IO clock (higher clock usually better). If an impossible rates combination is attempted the topology build will fail.

Reply:
my question was : can DMIC0 run at 16kHz and DMIC1 at 96kHz? In other words is the selection arbitrary? IIRC there was a second stage FIR, is this enabled for both DMIC0 and DMIC1?

Reply:
Yes, DMIC0 as 16 kHz and DMIC1 as 96 kHz is OK. The FIR decimators are in parallel and share same CIC, so they can be set independently to any rate as long as there is a suitable common CIC output rate. There's no option to place DMIC1 FIR after DMIC0 FIR even if it might have been a good idea for efficiency.

Reply:
Ah ok, thanks for educating me. I thought the two parts were serial with a 1/N downsampler in the second DMIC1.